airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -962,6 +963,19 @@ public Optional<ActorCatalog> getActorCatalog(final UUID actorId,
 
   }
 
+  public Optional<ActorCatalogFetchEvent> getMostRecentActorCatalogFetchEventForSource(final UUID sourceId) throws IOException {
+
+    final Result<Record> records = database.query(ctx -> ctx.select(ACTOR_CATALOG_FETCH_EVENT.asterisk())
+        .from(ACTOR_CATALOG_FETCH_EVENT)
+        .where(ACTOR_CATALOG_FETCH_EVENT.ACTOR_ID.eq(sourceId))
+        .orderBy(ACTOR_CATALOG_FETCH_EVENT.CREATED_AT.desc()).limit(1).fetch());
+
+    if (records.size() >= 1) {
```

Comment:
@benmoriceau findFirst returns a record instead of an ActorCatalogFetchEvent - this is the pattern we're using throughout the rest of the config repository so I used it here too

Reply:
About the suggestion, the complete solution should be
```return records.stream().findFirst().map(DbConverter::buildActorCatalogFetchEvent);```

The removes the need of the if and should be the expected nullable.

Reply:
ok updated throughout this file